api/v1alpha1/dataplane_konnect_extension_types.go

```diff
@@ -0,0 +1,81 @@
+package v1alpha1
+
+/*
+Copyright 2024 Kong Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&DataPlaneKonnectExtension{}, &DataPlaneKonnectExtensionList{})
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:categories=kong;all
+// +kubebuilder:subresource:status
+
+// DataPlaneKonnectExtension is the Schema for the dataplanekonnectextension API,
+// and is intended to be referenced as extension by the dataplane API.
+// If a DataPlane successfully refers a DataPlaneKonnectExtension, the DataPlane
+// deployment spec gets customized to include the konnect-related configuration.
+type DataPlaneKonnectExtension struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DataPlaneKonnectExtensionSpec   `json:"spec,omitempty"`
+	Status DataPlaneKonnectExtensionStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// DataPlaneKonnectExtensionList contains a list of DataPlaneKonnectExtension.
+type DataPlaneKonnectExtensionList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []DataPlaneKonnectExtension `json:"items"`
+}
+
+// DataPlaneKonnectExtensionSpec defines the desired state of DataPlaneKonnectExtension.
+type DataPlaneKonnectExtensionSpec struct {
+	// ControlPlaneID is the identifier of the Konnect Control Plane.
+	// +kubebuilder:validation:Required
+	ControlPlaneID string `json:"controlPlaneID"`
```

Comment:
Given that we're working on `KonnectControlPlane`, do you think it would make sense to pause for a bit with this approach and rework it, so that it supports different types of CP, e.g. instead of referencing by ID we could also reference by NN, thus having all the resources managed within the k8s cluster. 

WDYT?

Reply:
I think this is a good point. I'll capture it into the design doc I'm writing 👍